### PR TITLE
Testing and updating the Python Code Quick Start

### LINKED ISSUE #1350: [Site Bug] Add more detail to Python Quick Start /timescaledb/latest/quick-start/python/
# Describe the bug

In the Python Quick Start, give more explanation. Include the result, including how to see the table, and how to print it?

(Edited by Lana for clarity)

## What should the docs say?

# Page affected

[Provide the URL to the page you noticed the bug on]

# Version affected

[Which version of the site you noticed the bug on]

# Subject matter expert (SME)

@zseta @ryanbooz 

# Screenshots

[Attach images of screenshots showing the bug]

# Any further info

[Anything else you want to add, or further links]

### PR DESCRIPTION
**Pull request by Loquacity**
# Description

This is the first Quick Start to be refactored. Based on the changes we make to this one, we will develop a template for refactoring the others.

# Links

Partial: https://github.com/timescale/docs/issues/1187
Fixes: https://github.com/timescale/docs/issues/1350

# Writing help

For information about style and word usage, see the [style guide](https://docs.timescale.com/timescaledb/latest/contribute-to-docs/)

# Review checklists
Reviewers: use this section to ensure you have checked everything before approving this PR:

## Subject matter expert (SME) review checklist

- [ ] Is the content technically accurate?
- [ ] Is the content complete?
- [ ] Is the content presented in a logical order?
- [ ] Does the content use appropriate names for features and products?
- [ ] Does the content provide relevant links to further information?

## Documentation team review checklist

- [ ] Is the content free from typos?
- [ ] Does the content use plain English?
- [ ] Does the content contain clear sections for concepts, tasks, and references?
- [ ] Have any images been uploaded to the correct location, and are resolvable?
- [ ] If the page index was updated, are redirects required
      and have they been implemented?
- [ ] Have you checked the built version of this content?

### CODE REVIEW

File: timescaledb/quick-start/python.md [hunk 1]
```diff
@@ -4,421 +4,461 @@ excerpt: Get started with TimescaleDB for a Python application
...
+advanced TimescaleDB tutorials:
 
-- [Time Series Forecasting using TimescaleDB, R, Apache MADlib and Python][time-series-forecasting]
-- [Continuous Aggregates][continuous-aggregates]
-- [Try Other Sample Datasets][other-samples]
-- [Migrate your own Data][migrate]
+*   [Time Series Forecasting using TimescaleDB, R, Apache MADlib and Python][time-series-forecasting]
+*   [Continuous Aggregates][continuous-aggregates]
+*   [Try Other Sample Datasets][other-samples]
+*   [Migrate your own Data][migrate]
```
Comment (charislam) [inline timescaledb/quick-start/python.md:454]:
```suggestion
*   [Migrate Your own Data][migrate]
```

File: timescaledb/quick-start/python.md [hunk 2]
```diff
@@ -433,3 +473,8 @@ tutorials:
 [self-hosted-install]: /install/latest/self-hosted/
 [time-series-forecasting]: /timescaledb/:currentVersion:/tutorials/time-series-forecast/
 [virtual-env]: https://docs.python.org/3/library/venv.html
+[connect]: /timescaledb/:currentVersion:/quick-start/python/#connect-to-timescaledb
+[create-table]: /timescaledb/:currentVersion:/quick-start/python/#create-a-relational-table
+[create-hypertable]: /timescaledb/:currentVersion:/quick-start/python/#create-hypertable
+[insert]: /timescaledb/:currentVersion:/quick-start/python/#insert-rows-of-data
+[query]: /timescaledb/:currentVersion:/quick-start/python/#execute-a-query
```
Comment (charislam) [inline timescaledb/quick-start/python.md:480]: For anchor links on the same page, I would ditch the page path, so if you rename the page you don't have to rewrite them all. But it doesn't really matter

File: timescaledb/quick-start/python.md [hunk 3]
```diff
@@ -4,421 +4,461 @@ excerpt: Get started with TimescaleDB for a Python application
...
-```python
-import psycopg2
-```
+## Connect to TimescaleDB
 
-### Step 2: Compose a connection string
+In this section, you create a connection to TimescaleDB using the `psycopg2`
+library. This library is one of the most popular PostgreSQL libraries for
+Python. It allows you to execute raw SQL queries efficiently and safely, and
+prevents common attacks such as SQL injection. You don't need to use `psycopg2`,
```
[the remark this answers is not in the document]
Comment (mathisve) [reply timescaledb/quick-start/python.md:42]: Yes, that's the intended meaning. I see how it's confusing, I'll try to come up with some better wording

File: timescaledb/quick-start/python.md (hunk 2 above)
Comment (Loquacity) [reply timescaledb/quick-start/python.md:480]: So, just start with the # then? Or from the `/python/` ?

Comment (charislam) [reply timescaledb/quick-start/python.md:480]: With the `#`

File: _partials/_quickstart-intro.mdx [hunk 4]
```diff
@@ -0,0 +1,4 @@
+Timescale Code Quick Start Guides are designed to help you integrate TimescaleDB
+into your own programs. They use your favourite programming language to explain
```
[the remark this answers is not in the document]
Comment (Loquacity) [reply _partials/_quickstart-intro.mdx:2]: yep. Thanks for spotting this!

File: timescaledb/quick-start/python.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (Loquacity) [reply timescaledb/quick-start/python.md:42]: I found it a bit confusing when I was editing as well. I sort of thought that if you don't *need* to use it, then why are we? Which I think is answered (in a roundabout way) with the sentence about them "lacking some features", but it's all a little too obtuse for my liking.

**Comment by Loquacity** [comment]
If we put the Collapsible inside the procedure tags (which is where we usually put the heading for a procedure, it works pretty well, but we lose the styling on the numbers:

image
image

**Comment by Loquacity** [comment]
OK, every procedure is now collapsible, go check it out: http://docs-dev.timescale.com/docs-pythonqs-refactor-lana/timescaledb/pythonqs-refactor-lana/quick-start/python/

@charislam is there a way we can make some of these default to collapsed rather than expanded?

File: timescaledb/quick-start/python.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (mathisve) [reply timescaledb/quick-start/python.md]: I agree. Removed the last couple sentences.

**Comment by charislam** [comment]
> @charislam is there a way we can make some of these default to collapsed rather than expanded on page load?

Yeah, `<Collapsible defaultExpanded={false} />`

Unrelated, but when I was looking at the preview just now, the `virtual-env` link wasn't working, might want to double-check that you included the ref link for that!

**Comment by charislam** [comment]
Also re: collapsible styling I can probably fix that, let me take a look at it later today

**Review by mathisve — APPROVED** [review]

**Comment by Loquacity** [comment]
> Unrelated, but when I was looking at the preview just now, the `virtual-env` link wasn't working, might want to double-check that you included the ref link for that!

It's definitely in there, and it's leading to https://docs.python.org/3/library/venv.html, which is up. So I think it's OK?

**Comment by Loquacity** [comment]
@mathisve I've updated the collapsible sections so that, where we have more than one procedure in a section, the first procedure is expanded, but the subsequent procedures are collapsed by default. Can you take a look at http://docs-dev.timescale.com/docs-pythonqs-refactor-lana/timescaledb/pythonqs-refactor-lana/quick-start/python/ (in an incognito window), and tell me if you're OK with that? If you are, I think we can go ahead and merge 👍

**Comment by Loquacity** [comment]
Screenshot of collapsed (by default) second procedure in a section:

image

**Review by mathisve — APPROVED** [review]
Looks great! Absolutely love the collapsible procedures!

**Comment by charislam** [comment]
> > Unrelated, but when I was looking at the preview just now, the `virtual-env` link wasn't working, might want to double-check that you included the ref link for that!
> 
> It's definitely in there, and it's leading to https://docs.python.org/3/library/venv.html, which is up. So I think it's OK?

This is the one that's broken in preview:
image

Otherwise LGTM

**Comment by Loquacity** [comment]
> > > Unrelated, but when I was looking at the preview just now, the `virtual-env` link wasn't working, might want to double-check that you included the ref link for that!
> > 
> > 
> > It's definitely in there, and it's leading to https://docs.python.org/3/library/venv.html, which is up. So I think it's OK?
> 
> This is the one that's broken in preview: image
> 
> Otherwise LGTM

Oh! Is it the optional that's killing it?

**Comment by charislam** [comment]
> Oh! Is it the optional that's killing it?

Well, that's annoying. Did some experimenting, and it only happens if the optional is the first thing in the list item. If there's text before that, then it will parse it fine. (Probably a bug in our MDX parsing plugin.)

I can investigate a better solution but for now this will work as a hack. Just put an empty link right in front of the Optional:

```mdx
[](#)<Optional />A [Python virtual environment][virtual-env].
```

File: timescaledb/quick-start/python.md [hunk 5]
```diff
@@ -4,432 +4,485 @@ excerpt: Get started with TimescaleDB for a Python application
...
-* TimescaleDB installed, either in a [self-hosted environment][self-hosted-install] or [in the cloud][cloud-install]
-* The `psycopg2` library installed, [which you can install with pip][psycopg2-docs].
-* Optionally, a [Python virtual environment][virtual-env].
 
-## Connect Python to TimescaleDB
+*   Installed TimescaleDB. For more information, see the
+    [installation documentation][install].
+*   Installed the `psycopg2` library. For more information, see the
+    [psycopg2 documentation][psycopg2-docs].
+*   <Optional /> A [Python virtual environment][virtual-env].
```
Comment (Loquacity) [inline timescaledb/quick-start/python.md:29]:
```suggestion
*   [](#)<Optional />A [Python virtual environment][virtual-env].
```

**Comment by Loquacity** [comment]
> > Oh! Is it the optional that's killing it?
> 
> Well, that's annoying. Did some experimenting, and it only happens if the optional is the first thing in the list item. If there's text before that, then it will parse it fine. (Probably a bug in our MDX parsing plugin.)
> 
> I can investigate a better solution but for now this will work as a hack. Just put an empty link right in front of the Optional:
> 
> ```gfm
> [](#)<Optional />A [Python virtual environment][virtual-env].
> ```

Working now:

image

**Pull request merged**